RecoTracker/LSTCore/src/alpaka/Event.dev.cc

```diff
@@ -901,9 +906,11 @@ void Event::createPixelQuintuplets() {
   }
   if (trackCandidatesInGPU == nullptr) {
     trackCandidatesInGPU = new TrackCandidates();
-    trackCandidatesBuffers = new TrackCandidatesBuffer<Device>(
-        n_max_nonpixel_track_candidates + n_max_pixel_track_candidates, devAcc, queue);
-    trackCandidatesInGPU->setData(*trackCandidatesBuffers);
+    trackCandidatesDC_ =
+        new TrackCandidatesDeviceCollection(n_max_nonpixel_track_candidates + n_max_pixel_track_candidates, queue);
```

Comment:
If I trust gdb line numbers, the crash in the CPU backend case is here and says this is 
`PortableHostCollection<lst::TrackCandidatesSoALayout<128ul, false> >::PortableHostCollection<alpaka::QueueGenericThreadsBlocking<alpaka::DevCpu>, void>`

`print this->buffer_._M_payload._M_payload._M_value.m_spBufCpuImpl
$9 = {<std::__shared_ptr<alpaka::detail::BufCpuImpl<std::byte, std::integral_constant<unsigned long, 1>, unsigned int>, (__gnu_cxx::_Lock_policy)2>> = {<std::__shared_ptr_access<alpaka::detail::BufCpuImpl<std::byte, std::integral_constant<unsigned long, 1>, unsigned int>, (__gnu_cxx::_Lock_policy)2, false, false>> = {<No data fields>}, _M_ptr = 0x596d3d00, _M_refcount = {
      _M_pi = 0x596d3cf0}}, <No data fields>}`

this smells more like an ODR issue.

@makortel what could be the a good way to resolve this?
Please advise.

Reply:
after adding 
`ASSERT_DEVICE_MATCHES_HOST_COLLECTION(lst::TrackCandidatesDeviceCollection, lst::TrackCandidatesHostCollection);` in `TrackCandidate.h` I at least get a more consistent value to lead to an assert.

While in the example above `_M_ptr = 0x596d3d00` was aligned to 128 and the assert didn't make sense,
now I'm getting `_M_ptr = 0x597c3c40` which is not divisible by 128

Reply:
switching to `using TrackCandidatesSoA = TrackCandidatesSoALayout<cms::soa::CacheLineSize::defaultSize, cms::soa::AlignmentEnforcement::enforced>;` didn't help; now it crashes in `lst::TrackCandidatesSoALayout<128ul, true>::organizeColumnsFromBuffer` with `std::runtime_error` in `TrackCandidatesSoALayout::TrackCandidatesSoALayout: misaligned buffer`

Reply:
@slava77 I didn't see anything obviously wrong in the code. Do I understand correctly from the logs linked [above](https://github.com/SegmentLinking/cmssw/pull/90#issuecomment-2317217937) the problem occurs in the standalone setup? If that is the case, do you compile the Alpaka-dependent code with `-DALPAKA_DEFAULT_HOST_MEMORY_ALIGNMENT=128` (https://github.com/cms-sw/cmsdist/blob/IB/CMSSW_14_1_X/master/scram-tools.file/tools/alpaka/alpaka.xml)?

Reply:
> do you compile the Alpaka-dependent code with `-DALPAKA_DEFAULT_HOST_MEMORY_ALIGNMENT=128`

evidently we were not.
It seems to run after the update.
Thank you.